src/main/java/seedu/address/logic/commands/StartCommand.java

```diff
@@ -27,55 +24,59 @@
             + "Example: " + COMMAND_WORD + " physics";
 
+    public static final String MESSAGE_NO_FLASHCARDS = "No flashcards to test!";
+
     public static final String MESSAGE_START_TEST_SUCCESS = "Starting test...";
 
     private final AddressBookParser addressBookParser;
 
-    private final String deckName; // TODO: will integrate after deck class is completed
+    private final String tagName;
 
     public StartCommand(AddressBookParser addressBookParser) {
-        this.deckName = "Test"; // TODO: will get a random deck name from list of decks
+        this.tagName = "";
         this.addressBookParser = addressBookParser;
     }
 
-    public StartCommand(AddressBookParser addressBookParser, String deckName) {
-        this.deckName = deckName;
+    public StartCommand(AddressBookParser addressBookParser, String tagName) {
+        assert(!tagName.isEmpty());
+        this.tagName = tagName;
         this.addressBookParser = addressBookParser;
     }
 
     @Override
     public CommandResult execute(Model model) {
         requireNonNull(model);
-
-        // TODO: get the system to enter start mode, disable other commands except good/bad/end
-        addressBookParser.startTest();
-
-        // start stub
-        List<FlashCard> testList = new ArrayList<>();
-        testList.add(new FlashCard(new Question("1+1"), new Answer("2"), new Rating("good"),
-                SampleDataUtil.getTagSet("test")));
-        testList.add(new FlashCard(new Question("1+2"), new Answer("3"), new Rating("good"),
-                SampleDataUtil.getTagSet("test")));
-        testList.add(new FlashCard(new Question("1+3"), new Answer("4"), new Rating("good"),
-                SampleDataUtil.getTagSet("test")));
-        // end stub
-
-        // TODO: integrate into GUI
-        for (FlashCard fc : testList) {
-            fc.getQuestion();
-            // TODO: await command response
-            fc.getAnswer();
+        ArrayList<FlashCard> testList = searchTag(model);
+        model.initializeTestModel(testList);
+        if (!model.hasTestFlashCard()) {
+            return new CommandResult(MESSAGE_NO_FLASHCARDS);
         }
-
-        return new CommandResult(MESSAGE_START_TEST_SUCCESS);
-
+        addressBookParser.startTest();
+        String question = model.getTestQuestion();
+        addressBookParser.setAwaitingAnswer(true);
+        return new CommandResult(MESSAGE_START_TEST_SUCCESS + "\n" + question);
     }
 
     @Override
     public boolean equals(Object other) {
         return other == this // short circuit if same object
                 || (other instanceof StartCommand // instanceof handles nulls
-                && deckName.equals(((StartCommand) other).deckName)); // state check
+                && tagName.equals(((StartCommand) other).tagName)); // state check
     }
 
+    /** Searches the list of flashcard to fetch the relevant tags. */
+    private ArrayList<FlashCard> searchTag(Model model) {
```

Comment:
This may not be the best implementation. It will result in a change to the model's FilteredList, which I'd argue is an undesirable side-effect, since it's not what a user would expect from a command `start`. I recently added a method to the `Model` that can handle this for you - feel free to use it once my pull request is merged.